spring-cloud-config-server/src/main/java/org/springframework/cloud/config/server/environment/JGitEnvironmentRepository.java

```diff
@@ -161,6 +177,14 @@ public void setForcePull(boolean forcePull) {
 		this.forcePull = forcePull;
 	}
 
+	public boolean isDeleteUntrackedBranches() {
+		return deleteUntrackedBranches;
```

Comment:
Have you tested that this actually gets set when you set the property to `true`?  In `master` there is a configuration properties object and `deleteUntrackedBranches` is set in the constructor.  I am concerned that this is not the case here.  Maybe we should add an integration test?

Reply:
As for flow of setting this property:
It's a bit tricky, since whole ```JGitEnvironmentRepository``` is ```ConfigurationProperties```, for this you can take a look on ```MultipleJGitEnvironmentRepository``` which is inheriting this class, so properties set by ```ConfigurationPropertiesBindingPostProcessor```.
As for tests:
you can take a look on PR class ```JGitEnvironmentRepositoryIntegrationTests```. I've added two tests: ```testShouldReturnEnvironmentFromLocalBranchInCaseRemoteDeleted``` with unset property and exact the same test but with set property - ```testShouldFailIfRemoteBranchWasDeleted```